### PR TITLE
US8001 - For Developers Section

### PR DESCRIPTION
**Pull request by boggsey**
This adds the routes, content, and navigation for the For Developers section.

### CODE REVIEW

**Comment by zarahzachz** [comment]
@boggsey Looks good overall. We need some cleanup work done on `/developers/custom-npm-packages` though.

I talked with @tcmacdonald about this, and the 4 bullet points you have listed should actually be sidebar navigation with implementation details in the main body area. Some of the main content may not be written yet, but this is an example of the copy for the `content-blocks` section: https://design-int.crossroads.net/developers/ng2-components

File: src/app/developers/developers-routing.module.ts [hunk 1]
```diff
@@ -3,20 +3,110 @@ import { RouterModule, Routes } from '@angular/router';
...
+import { CodingStyleComponent } from './coding-style/coding-style.component';
+import { ContributionTipsComponent } from './contribution-tips/contribution-tips.component';
+import { CssSassComponent } from './css-sass/css-sass.component';
+
+import { CustomNpmPackagesComponent } from './custom-npm-packages/custom-npm-packages.component';
+import { OverviewDryComponent } from './custom-npm-packages/overview-dry/overview-dry.component';
+import { ContentBlocksComponent } from './custom-npm-packages/content-blocks/content-blocks.component';
+import { ReactiveSsoComponent } from './custom-npm-packages/reactive-sso/reactive-sso.component';
+import { SharedHeaderComponent } from './custom-npm-packages/shared-header/shared-header.component';
+import { CustomNpmPackagesJumbotronComponent } from './custom-npm-packages/custom-npm-packages-jumbotron/custom-npm-packages-jumbotron.component';
```
Comment (seancdavis) [inline src/app/developers/developers-routing.module.ts]: Error.

File: src/app/developers/developers.module.ts [hunk 2]
```diff
@@ -6,7 +6,26 @@ import { ContentBlockModule } from 'crds-ng2-content-block';
...
+import { CodingStyleComponent } from './coding-style/coding-style.component';
+import { ContributionTipsComponent } from './contribution-tips/contribution-tips.component';
+import { CssSassComponent } from './css-sass/css-sass.component';
+
+import { CustomNpmPackagesComponent } from './custom-npm-packages/custom-npm-packages.component';
+import { OverviewDryComponent } from './custom-npm-packages/overview-dry/overview-dry.component';
+import { ContentBlocksComponent } from './custom-npm-packages/content-blocks/content-blocks.component';
+import { ReactiveSsoComponent } from './custom-npm-packages/reactive-sso/reactive-sso.component';
+import { SharedHeaderComponent } from './custom-npm-packages/shared-header/shared-header.component';
+import { CustomNpmPackagesJumbotronComponent } from './custom-npm-packages/custom-npm-packages-jumbotron/custom-npm-packages-jumbotron.component';
```
Comment (seancdavis) [inline src/app/developers/developers.module.ts]: Error.

File: src/app/developers/developers.module.ts [hunk 3]
```diff
@@ -17,7 +36,24 @@ import { Ng2ComponentsComponent } from './ng2-components/ng2-components.componen
...
+    BuiltOnBootstrapComponent,
+    CodingStyleComponent,
+    ContributionTipsComponent,
+    CssSassComponent,
+    CustomNpmPackagesComponent,
+    OverviewDryComponent,
+    ContentBlocksComponent,
+    ReactiveSsoComponent,
+    SharedHeaderComponent,
+    CustomNpmPackagesJumbotronComponent,
```
Comment (seancdavis) [inline src/app/developers/developers.module.ts]: Error.

**Pull request merged**